src/UsgsAstroSarSensorModel.cpp

```diff
@@ -825,9 +926,41 @@ vector<csm::RasterGM::SensorPartials> UsgsAstroSarSensorModel::computeAllSensorP
     double* achievedPrecision,
     csm::WarningList* warnings) const
 {
-  return vector<csm::RasterGM::SensorPartials>(NUM_PARAMETERS, csm::RasterGM::SensorPartials(0.0, 0.0));
+  MESSAGE_LOG("Computing computeAllSensorPartials for ground point {}, {}, {} with desired precision {}",
+                              groundPt.x, groundPt.y, groundPt.z, desiredPrecision)
+   csm::ImageCoord imagePt = groundToImage(
+      groundPt, desiredPrecision, achievedPrecision, warnings);
+
+   return computeAllSensorPartials(
+      imagePt, groundPt, pSet, desiredPrecision, achievedPrecision, warnings);
+}
+
+std::vector<csm::RasterGM::SensorPartials> UsgsAstroSarSensorModel::computeAllSensorPartials(
+   const csm::ImageCoord& imagePt,
+   const csm::EcefCoord&  groundPt,
+   csm::param::Set        pSet,
+   double                 desiredPrecision,
+   double*                achievedPrecision,
+   csm::WarningList*      warnings) const
+{
+
+   MESSAGE_LOG("Computing computeAllSensorPartials for image {} {} and ground {}, {}, {} with desired precision {}",
+                                imagePt.line, imagePt.samp, groundPt.x, groundPt.y, groundPt.z, desiredPrecision)
+
+   std::vector<int> indices = getParameterSetIndices(pSet);
+   size_t num = indices.size();
+   std::vector<csm::RasterGM::SensorPartials> partials;
+   for (int index = 0; index < num; index++)
+   {
+     partials.push_back(
+         computeSensorPartials(
+            indices[index], imagePt, groundPt,
+            desiredPrecision, achievedPrecision, warnings));
+   }
+   return partials;
```

Comment:
We don't need this function, it's already in the RasterGM class from the CSM API.

Reply:
Does this make sense to pull out of linescan, then, too? (future ticket?)

Reply:
Yeah that sounds good as a future ticket.